### PR TITLE
OUP FTP error handling while download

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

File: hepcrawl/spiders/oup_spider.py [hunk 1]
```diff
@@ -151,6 +151,7 @@ def collect_files_to_download(self, host, ftp_folder):
 
     def download_files_from_ftp(self, ftp_folder):
         """"""
+        caught_error = False
```
Comment (drjova) [inline hepcrawl/spiders/oup_spider.py:154]: For retrying is better to use https://github.com/litl/backoff

File: hepcrawl/spiders/oup_spider.py [hunk 2]
```diff
@@ -192,13 +193,23 @@ def download_files_from_ftp(self, ftp_folder):
                 file_name = '%s_%s' % (
                     filename_prefix, os.path.basename(file_path))
                 local_filename = os.path.join(self.target_folder, file_name)
-                host.download(file_path, local_filename)
+                
+                try:
+                    host.download(file_path, local_filename)
+                except:
```
Comment (drjova) [inline hepcrawl/spiders/oup_spider.py:199]: Just a general comment :) It's not a good practice not to have an exception type, if we want to catch any exception (which is also not recommended) we should should use the following:

``` python
try:
...
except Exception:
...
```

File: hepcrawl/spiders/oup_spider.py [hunk 3]
```diff
@@ -192,12 +197,12 @@ def download_files_from_ftp(self, ftp_folder):
                 file_name = '%s_%s' % (
                     filename_prefix, os.path.basename(file_path))
                 local_filename = os.path.join(self.target_folder, file_name)
-                host.download(file_path, local_filename)
 
+                download_file_from_ftp()
```
Comment (drjova) [inline hepcrawl/spiders/oup_spider.py:201]: why this change? You can decorate a method. Also I would recommend be more specific in the Exception

Comment (ErnestaP) [reply hepcrawl/spiders/oup_spider.py:201]: It shouldn’t be class or function for a decorator?

Screen Shot 2022-02-21 at 15 58 18
